### PR TITLE
treeRows5: Use bits.TrailingZeros to find log2 of n

### PR DESCRIPTION
**Pull request by dergoegge**

### CODE REVIEW

**Comment by kcalvinalvin** [comment]
Tested. It is much faster indeed!

Though right now it doesn't handle the of when leaves are 0. I guess we can just put that in the comments since we really don't expect a tree with 0 leaves (and branching for checking if 0 reduces the speed by a lot).

Could you push the bits.TrailingZeros part (just need to replace the math.Log2() part) in my new code and add a comment about not using 0 as the function argument?

**Comment by dergoegge** [comment]
found a way to do it without the branching

**Comment by kcalvinalvin** [comment]
Oh great! One last thing but could you change second loop in TestTreeRows() so that it starts at 0? It's this code here
https://github.com/kcalvinalvin/utreexo/blob/e2e38d230f80994e71a534bdd73c8f5aee6cf193/accumulator/utils_test.go#L20-L21

I had it start from 1.

File: accumulator/utils_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 
 func TestTreeRows(t *testing.T) {
 	// Test all the powers of 2
-	for i := uint8(1); i <= 63; i++ {
+	for i := uint8(0); i <= 63; i++ {
```
Comment (kcalvinalvin) [inline accumulator/utils_test.go]: Oh this should be 1 since 1 << 0 is 1, which is not a power of 2

**Pull request merged**